### PR TITLE
Postgres/MySQL/MSSQL: Don't return connection failure details to the client

### LINKED ISSUE #22000: Hide IP address of Data Source server on error message
**What would you like to be added**:

image

Is it possible to hide IP address of Data Source server on error message?

**Why is this needed**:

for security reason

### LINKED ISSUE #26623: SSRF in grafana <=7.0.3
Grafana allows any user to add data sources. When a user puts the data source host and port he can save and test. 
For Mysql Data Source:
When a user puts host as localhost and a closed port it returns 
"dial tcp 127.0.0.1:20: connect: connection refused"

When a user puts an open port but which can't handle HTTP, returns
"commands out of sync. Did you run multiple statements at once?"

When a user puts an open port that handles HTTP requests it connects and didn't show any errors.
Also, other data source returns responses and attackers can find an open port using the data source response messages.

The issue is in the error messages, They are so straight forward as the response message/code.
That makes it vulnerable. By implementing a custom error message like "success" and "failed" will make it harder to guess the open port.
Hope it will help. If you need any more information feel free to ask.

### PR DESCRIPTION
**Pull request by marefr**
**What this PR does / why we need it**:
For security reasons, log any SQL connection error details rather than returning it to the client.

**Which issue(s) this PR fixes**:
Fixes #26623 
Fixes #22000 

**Special notes for your reviewer**:
- OpError in net package: https://github.com/golang/go/blob/d4b26382342c98a95b85140b2863bc30c48edd68/src/net/net.go#L435
- go-mssql connection error: https://github.com/denisenkom/go-mssqldb/blob/045585d74f9069afe2e115b6235eb043c8047043/tds.go#L904

### CODE REVIEW

**Review by dsotirakis — APPROVED** [review]
LGTM!

Two small comments:
* Maybe small unit tests could be added as well?
* Also, maybe take this into consideration since it touches similar parts https://github.com/grafana/grafana/issues/22000 (cc @ying-jeanne )

File: pkg/tsdb/mssql/mssql.go [hunk 1]
```diff
@@ -146,5 +148,14 @@ func (t *mssqlQueryResultTransformer) TransformQueryResult(columnTypes []*sql.Co
 }
 
 func (t *mssqlQueryResultTransformer) TransformQueryError(err error) error {
+	// go-mssql overrides source error why we currently match on string
```
Comment (aknuds1) [inline pkg/tsdb/mssql/mssql.go:151]: The current comment doesn't really parse, does this make sense?
```suggestion
	// go-mssql overrides source error, so we currently match on string
```

File: pkg/tsdb/mssql/mssql.go [hunk 2]
```diff
@@ -146,5 +148,14 @@ func (t *mssqlQueryResultTransformer) TransformQueryResult(columnTypes []*sql.Co
...
+	// ref https://github.com/denisenkom/go-mssqldb/blob/045585d74f9069afe2e115b6235eb043c8047043/tds.go#L904
+	if strings.HasPrefix(strings.ToLower(err.Error()), "unable to open tcp connection with host") {
+		t.log.Error("query error", "err", err)
+		return errConnectionFailed
+	}
+
 	return err
 }
+
+var errConnectionFailed = errors.New("failed to connect - please inspect Grafana server log for details")
```
Comment (aknuds1) [inline pkg/tsdb/mssql/mssql.go:161]: The same error is also defined in tsdb/sqleng, should we extract into _one_ definition?

**Review by aknuds1 — APPROVED** [review]
LGTM, but see comments.

File: pkg/tsdb/mssql/mssql.go (hunk 2 above)
Comment (marefr) [reply pkg/tsdb/mssql/mssql.go:161]: Good point

**Review by aknuds1 — APPROVED** [review]

**Comment by ying-jeanne** [comment]
LGTM, we might update a simple test for postgres_test.go, to make sure that connection error is captured by *net.OpError? If it is something easy to add.

**Review by ying-jeanne — APPROVED** [review]

**Comment by marefr** [comment]
@ying-jeanne the test in sqleng package already applies to all sql data sources. Or you referring to some other kind of test needed?

**Pull request merged**